analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -188,13 +188,14 @@ open class Npm(
                 val keyAndValue = line.split('=', limit = 2).map { it.trim() }
                 if (keyAndValue.size == 2) {
                     val (key, value) = keyAndValue
-                    val proxyUrl = value.takeIf { it.matches(HTTP_REGEX) } ?: when (key) {
-                        "proxy" -> "http://$value"
-                        "https-proxy" -> "https://$value"
-                        else -> ""
+                    val proxyUrl = value.takeIf { it.matches(HTTP_REGEX) } ?: when {
```

Comment:
I think the following two options would both be slightly better:
(1) make `when` black return non-null strings using empty leads to simpler handling (you can keep isNotEmpty)
(2) keep as is but use 'https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/is-null-or-empty.html' instead of the nullability check. It removes the hidden assumption that the URL is not empty.

Reply:
Please note that `proxyUrl` can never be empty with the current code, hence I don't think your option (2) applies. Similarly regarding optino (1), it does not make sense to again set `proxyUrl` to `""` in the error case as it would not help to reduce a "null or empty" check to just an "empty" check. or am I missing something?

Reply:
> Please note that `proxyUrl` can never be empty with the current code, hence I don't think your option (2) applies.

I was well aware of that fact when writing this and again, the code below the when make less assumptions about what the code above does which is good imo.

Regarding option my (1) I missed the takeIf so the variable can still be null. My underlying idea was to avoid the nullable var type.

Reply:
Actually, the `takeIf` itself is irrelevant here, as it's eventual `null` return value only triggers the `when` fallback.

But I actually like `proxyUrl` to be nullable now as I think `null` is a better value to indicate something invalid than `""`.

Reply:
> But I actually like `proxyUrl` to be nullable now as I think `null` is a better value to indicate something invalid than `""`.
The **nit** problem I see with this is that you don't have the choice to represent INVALID as null
but not as "". You can choose only betweem
- INVALID = "" || null
- INVALID = ""

At least when it comes to what the compiler can check for us. Thus choosing `null` implies you need to also still handle "" because "" is invalid in case of URLs.

Reply:
Again, I'd agree with you if `proxyUrl` could be `""`, but in the current code it can't. So is your point that you want to be prepared for possible future changes that actually *might* set `proxyUrl` to `""`, and avoid the need to adjust the condition for that?

Reply:
yes, my point is (1) the future change you mention and (2) less assumptions -> simpler

but I'm also fine with this as is, it's a not important detail and maybe personal preference.